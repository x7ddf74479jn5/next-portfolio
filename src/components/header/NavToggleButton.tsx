import CloseIcon from "@material-ui/icons/Close";
import MenuIcon from "@material-ui/icons/Menu";
import styles from "src/styles/components/header/HeaderButton.module.scss";
import type { ButtonType } from "src/types/utils";

type Props = ButtonType & {
  isOpen: boolean;
};

const NavToggleButton: React.VFC<Props> = ({ isOpen, onClick }) => {
  /* <!-- Navigation toggle button for Smartphone--> */

  const navToggleButtonStyle = isOpen ? styles.navToggleButtonOpen : styles.navToggleButton;

  return (
    <button
      className={navToggleButtonStyle}
      data-about-nav-toggle
      aria-label="ナビゲーション メニューの切り替え"
      id="nav-toggle-button"
      onClick={onClick}
    >
      {isOpen ? (
        <CloseIcon aria-label="ナビゲーション メニューを閉じる" />
      ) : (
        <MenuIcon aria-label='ナビゲーション メニューを開く"' />
      )}
    </button>
  );
};

export default NavToggleButton;
