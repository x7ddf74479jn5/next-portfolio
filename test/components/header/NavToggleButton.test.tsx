import userEvent from "@testing-library/user-event";
import NavToggleButton from "src/components/header/NavToggleButton";
import type { RenderResult } from "test/test-utils";
import { render } from "test/test-utils";

describe("NavToggleButton", () => {
  let renderResult: RenderResult;
  let mockHandleClick: jest.Mock<any, any>;

  describe("closed", () => {
    beforeEach(() => {
      mockHandleClick = jest.fn();
      renderResult = render(<NavToggleButton isOpen={false} onClick={mockHandleClick} />);
    });

    it("matches snapshot", () => {
      expect(renderResult.asFragment()).toMatchSnapshot();
    });

    it("rendered correctly", () => {
      const buttonElement = renderResult.getByRole("button");
      expect(buttonElement).toHaveAttribute("aria-label", expect.stringContaining("ナビゲーション"));
      expect(buttonElement).toHaveClass("navToggleButton");
      const imgElement = renderResult.container.getElementsByTagName("svg")[0];
      expect(imgElement).toHaveAttribute("aria-label", expect.stringContaining("開く"));
    });

    it("user Interaction", () => {
      userEvent.click(renderResult.getByRole("button"));
      expect(mockHandleClick).toBeCalled();
    });
  });

  describe("open", () => {
    beforeEach(() => {
      mockHandleClick = jest.fn();
      renderResult = render(<NavToggleButton isOpen={true} onClick={mockHandleClick} />);
    });

    it("matches snapshot", () => {
      expect(renderResult.asFragment()).toMatchSnapshot();
    });

    it("rendered correctly", () => {
      const buttonElement = renderResult.getByRole("button");
      expect(buttonElement).toHaveClass("navToggleButtonOpen");

      const imgElement = renderResult.container.getElementsByTagName("svg")[0];
      expect(imgElement).toHaveAttribute("aria-label", expect.stringContaining("閉じる"));
    });

    it("user Interaction", () => {
      userEvent.click(renderResult.getByRole("button"));
      expect(mockHandleClick).toBeCalled();
    });
  });
});
