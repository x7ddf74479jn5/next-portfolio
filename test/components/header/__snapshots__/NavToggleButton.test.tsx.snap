// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavToggleButton closed matches snapshot 1`] = `
<DocumentFragment>
  <button
    aria-label="ナビゲーション メニューの切り替え"
    class="navToggleButton"
    data-about-nav-toggle="true"
    id="nav-toggle-button"
  >
    <svg
      aria-hidden="true"
      aria-label="ナビゲーション メニューを開く\\""
      class="MuiSvgIcon-root"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
      />
    </svg>
  </button>
</DocumentFragment>
`;

exports[`NavToggleButton open matches snapshot 1`] = `
<DocumentFragment>
  <button
    aria-label="ナビゲーション メニューの切り替え"
    class="navToggleButtonOpen"
    data-about-nav-toggle="true"
    id="nav-toggle-button"
  >
    <svg
      aria-hidden="true"
      aria-label="ナビゲーション メニューを閉じる"
      class="MuiSvgIcon-root"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
      />
    </svg>
  </button>
</DocumentFragment>
`;
